include/simppl/any.h

```diff
@@ -1,406 +1,680 @@
 #ifndef SIMPPL_ANY_H
 #define SIMPPL_ANY_H
 
-
 #include <any>
+#include <bits/utility.h>
 #include <cassert>
+#include <cstdint>
 #include <cstring>
+#include <dbus/dbus-protocol.h>
+#include <iostream>
 #include <memory>
 #include <sstream>
+#include <stdexcept>
+#include <sys/types.h>
+#include <tuple>
+#include <utility>
 #include <variant>
 
 #include <dbus/dbus.h>
+#include <vector>
 
-#include "simppl/serialization.h"
 #include "simppl/detail/deserialize_and_return.h"
+#include "simppl/map.h"
+#include "simppl/pod.h"
+#include "simppl/serialization.h"
+#include "simppl/tuple.h"
+#include "simppl/type_mapper.h"
+#include "simppl/variant.h"
+#include "simppl/vector.h"
+
+namespace simppl {
+
+namespace dbus {
+struct Any;
+struct IntermediateAnyVec;
+struct IntermediateAnyMapElement;
+struct IntermediateAnyTuple;
+
+template <typename T> T decode2(DBusMessageIter &iter);
+template <> IntermediateAnyVec decode2(DBusMessageIter &iter);
+template <> IntermediateAnyMapElement decode2(DBusMessageIter &iter);
+template <> IntermediateAnyTuple decode2(DBusMessageIter &iter);
+template <> Any decode2(DBusMessageIter &iter);
+template <> std::string decode2(DBusMessageIter &iter);
+
+template <typename T> void encode2(DBusMessageIter &iter, const T &val);
+template <> void encode2(DBusMessageIter &iter, const IntermediateAnyVec &val);
+template <>
+void encode2(DBusMessageIter &iter, const IntermediateAnyMapElement &val);
+template <>
+void encode2(DBusMessageIter &iter, const IntermediateAnyTuple &val);
+template <> void encode2(DBusMessageIter &iter, const Any &val);
+template <> void encode2(DBusMessageIter &iter, const std::string &val);
 
+/**
+ * A true D-Bus variant that can really hold *anything*, as a replacement
+ * for variants.
+ **/
+struct Any {
+  // DBUS_TYPE_...
+  int containedType{DBUS_TYPE_INVALID};
+  std::string containedTypeSignature;
+  std::any value_;
+
+  Any(){};
+
+  Any(int type, std::string &&containedTypeSignature, std::any &&any)
```

Comment:
templatify to be able to use std::forward instead of only std::move

Reply:
Won't happen since it's the ctr used for the internal ctr and friends would make it too bloated in my eyes.